database/migrations/default/1744826895218_ems_remove_match_status_constraint/up.sql

```
@@ -0,0 +1,47 @@
+alter table ems__incidents
+drop constraint matched_crash_pks_non_null,
+add column created_by text not null default 'system',
+add column updated_by text not null default 'system';
+
+comment on column ems__incidents.matched_crash_pks is 'The IDs of multiple crashes that were found to match this record. Set via trigger.''';
+
+--
+-- Trigger function that sets the ems__incident `crash_pk` based on the
+-- provided person_id and updates the crash_pk of all EMS records that
+-- share the same incident number
+--
+create or replace function update_person_crash_ems_match()
+returns trigger as $$
+DECLARE
+    matching_person_record RECORD;
+BEGIN
+    IF NEW.person_id is null
+        THEN return null;
```

so by deleting / nulling out a `person_id`, the `crash_id` the ems record is linked to will remain instead of nulling out -- is that what we want to happen? should there be a way to null a crash link?

---

yeah, thanks for pointing this out. i left the behavior as-is because i wasn't sure what the desired behavior would be in different scenarios. but i guess if you're nullifying the person ID you probably want to nullify the crash ID as well, right?

---

Note to self—if we decide to change this block so that the trigger nullifies the `crash_pk` column as well, we need to check to make sure that `OLD.person_id` is not null.

---

why do we need to check that `OLD.person_id` was not null?

but yeah i do think it makes sense to be able to null out the `crash_id`, but im not sure if vz team would ever want to be able to null out person id but keep the crash id around. could be something that we fine tune once the vz team starts using this or maybe we can ask xavier?

---

> why do we need to check that OLD.person_id was not null?

Many EMS records will have a crash ID and  null person ID as a result of the crash-EMS trigger. So we need to make sure that the event that triggers this update is actually intending to nullify the person ID, and not the result of some other update that would unintentionally null out the crash ID. 

I'm with you that we need a mechanism to null out crash ID and it makes sense to do it in this trigger. I think it will just be one or two lines so maybe I'll push that up in this PR 👍